### PR TITLE
feature: #394 URL fragment

### PR DESCRIPTION
**Pull request by soujiro32167**

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/http/URL.scala [hunk 1]
```diff
@@ -97,4 +99,13 @@ object URL {
         else s"${scheme.asString}://$host:$port$path"
     }
   }
+
+  case class Fragment private (raw: String, decoded: String)
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/http/URL.scala:103]: I think we should only keep decoded value in the fragment.

**Comment by amitksingh1490** [comment]
can you please run command `fmt` on sbt shell, and push the changes it will fix the formatting.

File: zio-http/src/main/scala/zhttp/http/URL.scala (hunk 1 above)
Comment (soujiro32167) [reply zio-http/src/main/scala/zhttp/http/URL.scala:103]: Then we would have to re-encode it when going back to string...

Comment (soujiro32167) [reply zio-http/src/main/scala/zhttp/http/URL.scala:103]: If its an allocation concern, how about we keep it as a `() => decoded`?

Comment (soujiro32167) [reply zio-http/src/main/scala/zhttp/http/URL.scala:103]: Or the other way around - `case class Fragment private (encoded: () => String, decoded: String)`

**Comment by tusharmath** [comment]
We will look at the performance piece a bit later. Thank you for your contribution!

**Review by tusharmath — APPROVED** [review]

**Pull request merged**